mesonbuild/mtest.py

```diff
@@ -405,11 +405,32 @@ def print_stats(self, numlen, tests, name, result, i):
         num = '%s%d/%d' % (startpad, i + 1, len(tests))
         padding1 = ' ' * (38 - len(name))
         padding2 = ' ' * (8 - len(result.res.value))
-        result_str = '%s %s  %s%s%s%5.2f s' % \
-            (num, name, padding1, result.res.value, padding2, result.duration)
+        status = ''
+        if result.res is TestResult.FAIL:
+            # Note: We can't use `os.WIFSIGNALED(result.returncode)` and the
+            # related functions here because the status returned by subprocess
+            # is munged. It is supposed to return a negative value if the
+            # process was killed by a signal but in reality it returns a
+            # positive value with 128 added to the signal number. Which is the
+            # opposite of the usual UNIX exit status semantics. That is, a
+            # process killed by a signal typically has a returncode < 128 while
+            # a termination due to an `exit()` call has a returncode > 128. So
```

Comment:
Aren't these < and > characters the wrong way around in this particular sentence? They don't seem to match the text above or the code below. (that is, a process killed by a signal has a return code _larger_ than 128)

Reply:
I'm skeptical of the correctness of that whole comment .

Exit codes are unsigned integers (maybe even only 8 bit unsigned integers).  signals are not errnos. 

os.WTERMSIG/os.WIFSIGNALED should be the right tool to use here...

Reply:
Gah! I just took my own advice. The python3 `subprocess` class behaves as documented with respect to the `returncode` attribute when I use it from Ipython or a hand crafted python program. Which implies Meson is munging the subprocess `returncode`. I'm going to withdraw this PR while I ponder the Meson code to understand why my change produces the expected result even though it is at odds with the documented behavior of the underlying `subprocess` module.

Reply:
@jon-turney I just read your comment a second time and I don't understand what you mean by "signals are not errnos". Where in this PR did I suggest that errno values are relevant?

Reply:
Apologies, I hadn't noticed that sentence in the documentation for `subprocess`, and indeed [it does crack open the exit status for you](https://github.com/python/cpython/blob/3.6/Lib/subprocess.py#L1348-L1363), so you can just ignore what I wrote :smile:

Then I'd misunderstood what you wrote to say that wait() returns -signal, as some libc functions return -errno.

Reply:
> Which implies Meson is munging the subprocess returncode

There should be nothing in the Meson code that intentionally does that. If it happens, it's a bug.